cache.go

```diff
@@ -139,11 +144,22 @@ func (c *cache) remove(name string) error {
 	return nil
 }
 
+func (c *cache) monitor(tick time.Duration) {
+	ticker := time.NewTicker(tick)
+	for range ticker.C {
+		c.mu.RLock()
```

Comment:
This is a long time to hold this lock, even if it is a read lock. Maybe hold the lock, make a copy of entries, and then unlock while you do the fetches?